### DISCUSSION TITLE
How would I set the background color of an output SVG?

### DISCUSSION THREAD

**Discussion by Mee42** [JetBrains/lets-plot-kotlin#45] [Q&A] [answered]
It seems like the default is for the SVG to be transparent, which doesn't work very well with some svg renderers such as the one built into IntelliJ. Is there any way to change this?

 The code I'm using to test this is:
 ```kotlin
val df = mapOf(
    "display" to (0..10).map { (0..10).random() },
    "value" to (0..10).map { (3..35).random() }
)

val plot = lets_plot(df) {
    x = "display"
    y = "value"
} + geom_point(df)
ggsave(plot, "graph.svg")
```

The outputted svg looks like this when rendered in Firefox

![image](https://user-images.githubusercontent.com/39016114/102721377-6af0f200-42c8-11eb-9893-ecc131ef149c.png)

and in IntelliJ

![image](https://user-images.githubusercontent.com/39016114/102721390-78a67780-42c8-11eb-9b8e-34fb314b899a.png)

**Comment by alshan** [comment, marked as answer]
That's right, there is no 'background' in SVG container - it's always transparent.

One way to make it opaque is to manually add a filled rectangle before the rest of the contents:
```
<rect width="100%" height="100%" fill="white"/>
```
Such an option in `ggsave()` would be useful IMO.
Even better would be to have an easy way to set a desirable background in IDEA SVG viewer itself - please vote: https://youtrack.jetbrains.com/issue/IDEA-250345

**Reply by Mee42** [reply]
Seems like I misdiagnosed this issue - the SVG is transparent (as you said, all SVGs are transparent) but already contains
```xml
<rect class="plt-backdrop" width="100%" height="100%">
```
which refers back to some `<style>`
```css
.plt-backdrop {
   fill: white;
}
```
If I inline the `fill="white"` like you had in your example, it renders fine. With some other testing, it seems like IntelliJ just doesn't support CSS in SVG at all, which is unfortunate. Oh well.


Thank you for your help, though!

**Reply by alshan** [reply]
>  it seems like IntelliJ just doesn't support CSS in SVG at all

I didn't know this and this is unfortunate indeed, because we have few other, font-related styles in SVG. 

Thank you for letting us know! Maybe you can try raster format instead. You can choose scale / dpi to your preference there.
